<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="
    Functions to summarize DNA methylation data using regional principal
    components. Regional principal components are computed using principal 
    components analysis within genomic regions to summarize the variability 
    in methylation levels across CpGs. The number of principal components is
    chosen using either the Marcenko-Pasteur or Gavish-Donoho method to
    identify relevant signal in the data.">
<title>Summarizing Regional Methylation with Regional Principal Components Analysis • regionalpcs</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Summarizing Regional Methylation with Regional Principal Components Analysis">
<meta property="og:description" content="
    Functions to summarize DNA methylation data using regional principal
    components. Regional principal components are computed using principal 
    components analysis within genomic regions to summarize the variability 
    in methylation levels across CpGs. The number of principal components is
    chosen using either the Marcenko-Pasteur or Gavish-Donoho method to
    identify relevant signal in the data.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">regionalpcs</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/regionalpcs.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/tyeulalio/regionalpcs/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="regionalpcs">regionalpcs<a class="anchor" aria-label="anchor" href="#regionalpcs"></a>
</h1></div>
</div>
<div class="section level1">
<h1 id="table-of-contents">Table of Contents<a class="anchor" aria-label="anchor" href="#table-of-contents"></a>
</h1>
<ol style="list-style-type: decimal">
<li><a href="#introduction">Introduction</a></li>
<li><a href="#installation">Installation</a></li>
<li>
<a href="#regionalpcs-r-package-tutorial">regionalpcs R Package Tutorial</a>
<ul>
<li>3.1 <a href="#loading-required-packages">Loading Required Packages</a>
</li>
<li>3.2 <a href="#load-the-dataset">Load the Dataset</a>
<ul>
<li>3.2.1 <a href="#overview">Overview</a>
</li>
<li>3.2.2 <a href="#inspecting-the-data">Inspecting the Data</a>
</li>
</ul>
</li>
<li>3.3 <a href="#obtaining-methylation-array-probe-positions">Obtaining Methylation Array Probe Positions</a>
<ul>
<li>3.3.1 <a href="#introduction-1">Introduction</a>
</li>
<li>3.3.2 <a href="#extract-probe-names-and-positions">Extract Probe Names and Positions</a>
</li>
<li>3.3.3 <a href="#load-illuminaprobe-positions">Load Illumina 450k Array Probe Positions</a>
</li>
<li>3.3.4 <a href="#merge-data-frames">Merge Data Frames</a>
</li>
<li>3.3.5 <a href="#addressing-genome-build-discrepancies">Addressing Genome Build Discrepancies</a>
</li>
</ul>
</li>
<li>3.4 <a href="#processing-and-filtering-methylation-data">Processing and Filtering Methylation Data</a>
<ul>
<li>3.4.1 <a href="#introduction-2">Introduction</a>
</li>
<li>3.4.2 <a href="#remove-low-variance-cpgs">Remove Low Variance CpGs</a>
</li>
<li>3.4.3 <a href="#normalize-methylation-values">Normalize Methylation Values</a>
</li>
</ul>
</li>
<li>3.5 <a href="#summarizing-gene-region-types">Summarizing Gene Region Types</a>
<ul>
<li>3.5.1 <a href="#introduction-3">Introduction</a>
</li>
<li>3.5.2 <a href="#load-gene-region-annotations">Load Gene Region Annotations</a>
</li>
<li>3.5.3 <a href="#create-a-region-map">Create a Region Map</a>
<ul>
<li>3.5.3.1 <a href="#extract-cpg-positions">Extract CpG Positions</a>
</li>
<li>3.5.3.2 <a href="#convert-to-genomicranges-and-find-overlaps">Convert to GenomicRanges and Find Overlaps</a>
</li>
</ul>
</li>
<li>3.5.4 <a href="#summarizing-gene-regions">Summarizing Gene Regions with Regional Principal Components</a>
<ul>
<li>3.5.4.1 <a href="#compute-regional-pcs">Compute Regional PCs</a>
</li>
<li>3.5.4.2 <a href="#inspecting-the-output">Inspecting the Output</a>
</li>
<li>3.5.4.3 <a href="#extracting-and-viewing-regional-pcs">Extracting and Viewing Regional PCs</a>
</li>
<li>3.5.4.4 <a href="#understanding-the-results">Understanding the Results</a>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</li>
</ol>
</div>
<div class="section level1">
<h1 id="regionalpcs-1">regionalpcs<a class="anchor" aria-label="anchor" href="#regionalpcs-1"></a>
</h1>
<p>Tiffany Eulalio</p>
<p>The <code>regionalpcs</code> package aims to address the challenge of summarizing and interpreting DNA methylation data at a regional level. Traditional methods of analysis may not capture the biological complexity of methylation patterns, potentially leading to less accurate or less meaningful interpretations. This package introduces the concept of regional principal components (rPCs) as a tool for capturing more biologically relevant signals in DNA methylation data. By using rPCs, researchers can gain new insights into complex interactions and effects in methylation data that might otherwise be missed.</p>
</div>
<div class="section level1">
<h1 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h1>
<p>You can install the regionalpcs package from Bioconductor using the following command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"regionalpcs"</span><span class="op">)</span></span></code></pre></div>
<p>You can install the development version of regionalpcs from GitHub with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install devtool package if needed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># download the regionalpcs package</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"tyeulalio/regionalpcs"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level1">
<h1 id="regionalpcs-r-package-tutorial">
<code>regionalpcs</code> R Package Tutorial<a class="anchor" aria-label="anchor" href="#regionalpcs-r-package-tutorial"></a>
</h1>
<div class="section level2">
<h2 id="loading-required-packages">Loading Required Packages<a class="anchor" aria-label="anchor" href="#loading-required-packages"></a>
</h2>
<p>This tutorial depends on several Bioconductor packages. These packages should be loaded at the beginning of the analysis.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/tyeulalio/regionalpcs" class="external-link">regionalpcs</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">RNOmni</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">IlluminaHumanMethylation450kanno.ilmn12.hg19</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://www.bioconductor.org/help/workflows/liftOver/" class="external-link">liftOver</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<p>Here, we load the regionalpcs package, which is the main package we’ll be using in this tutorial. We also load RNOmni, which provides normalization functions, GenomicRanges, which provides tools for working with genomic intervals, and tidyverse, which provides a suite of tools for data manipulation and visualization.</p>
<p>It’s important to note that you need to have these packages installed on your machine before loading them. You can install them using the install.packages() function in R.</p>
<p>Once the packages are loaded, we can start using the functions provided by each package.</p>
</div>
<div class="section level2">
<h2 id="load-the-dataset">Load the dataset<a class="anchor" aria-label="anchor" href="#load-the-dataset"></a>
</h2>
<div class="section level3">
<h3 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<p>The betas dataset in the <code>regionalpcs</code> package is a subset of 450k array methylation data from TCGA, containing 293 methylation sites and 300 samples. We’ll load this data into our R session and explore its structure.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"betas"</span>, package <span class="op">=</span> <span class="st">"regionalpcs"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="inspecting-the-data">Inspecting the Data<a class="anchor" aria-label="anchor" href="#inspecting-the-data"></a>
</h3>
<p>We can take a quick look at the dimensions of the dataset and the first few rows to understand its structure.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">betas</span><span class="op">)</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span></span>
<span><span class="co">#&gt;                                     TCGA-EJ-7781-11A TCGA-BH-A1FE-11B</span></span>
<span><span class="co">#&gt; chr16_53434200_53434201_cg00000029        0.20361112       0.13654490</span></span>
<span><span class="co">#&gt; chr15_22838620_22838621_cg00000622        0.01311223       0.01024075</span></span>
<span><span class="co">#&gt; chr1_166989202_166989203_cg00001349       0.72180841       0.74037266</span></span>
<span><span class="co">#&gt; chr8_119416178_119416179_cg00002464       0.05881476       0.05834758</span></span>
<span><span class="co">#&gt; chr6_169751536_169751537_cg00005543       0.01868565       0.01808436</span></span>
<span><span class="co">#&gt; chr12_52069532_52069533_cg00006122        0.05748535       0.06136361</span></span>
<span><span class="co">#&gt;                                     TCGA-BH-A0C3-11A</span></span>
<span><span class="co">#&gt; chr16_53434200_53434201_cg00000029        0.12996001</span></span>
<span><span class="co">#&gt; chr15_22838620_22838621_cg00000622        0.01847991</span></span>
<span><span class="co">#&gt; chr1_166989202_166989203_cg00001349       0.76361838</span></span>
<span><span class="co">#&gt; chr8_119416178_119416179_cg00002464       0.06189946</span></span>
<span><span class="co">#&gt; chr6_169751536_169751537_cg00005543       0.02085631</span></span>
<span><span class="co">#&gt; chr12_52069532_52069533_cg00006122        0.05805733</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">betas</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 293 300</span></span></code></pre></div>
<p>Note that the row names are CpG IDs and genomic positions, and the columns contain methylation beta values ranging from 0 to 1 for individual samples.</p>
</div>
</div>
<div class="section level2">
<h2 id="obtaining-methylation-array-probe-positions">Obtaining Methylation Array Probe Positions<a class="anchor" aria-label="anchor" href="#obtaining-methylation-array-probe-positions"></a>
</h2>
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>To perform accurate and informative analyses on methylation array data, it is critical to have precise genomic positions for each probe. The <code>IlluminaHumanMethylation450kanno.ilmn12.hg19</code> package contains annotations for 450k methylation arrays, which can be utilized for this purpose. This section will walk you through the steps to associate each probe in your dataset with its genomic position.</p>
</div>
<div class="section level3">
<h3 id="extract-probe-names-and-positions">Extract Probe Names and Positions<a class="anchor" aria-label="anchor" href="#extract-probe-names-and-positions"></a>
</h3>
<p>First, we’ll extract the probe names from the betas data frame and use regular expressions to separate the CpG identifier from its genomic position.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract probe names and CpG positions from row names of 'betas'</span></span>
<span><span class="va">cpg_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>cpgs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">betas</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span><span class="va">cpgs</span>,</span>
<span>        into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"cpg_pos"</span>, <span class="st">"probe"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"_(?=[^_]+$)"</span>,</span>
<span>        extra <span class="op">=</span> <span class="st">"merge"</span></span>
<span>    <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cpg_df</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    cpg_pos      probe</span></span>
<span><span class="co">#&gt; 1  chr16_53434200_53434201 cg00000029</span></span>
<span><span class="co">#&gt; 2  chr15_22838620_22838621 cg00000622</span></span>
<span><span class="co">#&gt; 3 chr1_166989202_166989203 cg00001349</span></span>
<span><span class="co">#&gt; 4 chr8_119416178_119416179 cg00002464</span></span>
<span><span class="co">#&gt; 5 chr6_169751536_169751537 cg00005543</span></span>
<span><span class="co">#&gt; 6  chr12_52069532_52069533 cg00006122</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-illumina-450k-array-probe-positions">Load Illumina 450k Array Probe Positions<a class="anchor" aria-label="anchor" href="#load-illumina-450k-array-probe-positions"></a>
</h3>
<p>Next, let’s load the Illumina 450k array probe positions for further annotation.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">Locations</span><span class="op">)</span></span>
<span><span class="va">probe_positions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">Locations</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">probe_positions</span><span class="op">)</span></span>
<span><span class="co">#&gt;             chr      pos strand</span></span>
<span><span class="co">#&gt; cg00050873 chrY  9363356      -</span></span>
<span><span class="co">#&gt; cg00212031 chrY 21239348      -</span></span>
<span><span class="co">#&gt; cg00213748 chrY  8148233      -</span></span>
<span><span class="co">#&gt; cg00214611 chrY 15815688      -</span></span>
<span><span class="co">#&gt; cg00455876 chrY  9385539      -</span></span>
<span><span class="co">#&gt; cg01707559 chrY  6778695      +</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="merge-data-frames">Merge Data Frames<a class="anchor" aria-label="anchor" href="#merge-data-frames"></a>
</h3>
<p>Now, we merge the extracted probe names with the Illumina 450k array probe positions.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">formatted_probe_positions</span> <span class="op">&lt;-</span> <span class="va">probe_positions</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html" class="external-link">rownames_to_column</a></span><span class="op">(</span><span class="st">"probe"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">new_cpg_df</span> <span class="op">&lt;-</span> <span class="va">cpg_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">formatted_probe_positions</span>, by <span class="op">=</span> <span class="st">"probe"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">new_cpg_df</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    cpg_pos      probe   chr       pos strand</span></span>
<span><span class="co">#&gt; 1  chr16_53434200_53434201 cg00000029 chr16  53468112      +</span></span>
<span><span class="co">#&gt; 2  chr15_22838620_22838621 cg00000622 chr15  23034447      +</span></span>
<span><span class="co">#&gt; 3 chr1_166989202_166989203 cg00001349  chr1 166958439      -</span></span>
<span><span class="co">#&gt; 4 chr8_119416178_119416179 cg00002464  chr8 120428418      +</span></span>
<span><span class="co">#&gt; 5 chr6_169751536_169751537 cg00005543  chr6 170151632      +</span></span>
<span><span class="co">#&gt; 6  chr12_52069532_52069533 cg00006122 chr12  52463316      +</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="addressing-genome-build-discrepancies">Addressing Genome Build Discrepancies<a class="anchor" aria-label="anchor" href="#addressing-genome-build-discrepancies"></a>
</h3>
<p>It’s critical to ensure that the genome builds match across datasets. In this example, we’ll use the <code>GenomicRanges</code> and <code>liftOver</code> packages to convert the genomic positions from hg19 to hg38. Here’s a quick example on how to lift over positions from one build to another. <strong>Always ensure that you are working with the correct genome build and that the build matches across all your datasets, or else you will run into big issues!</strong></p>
<p>We need a chain file to lift the genomic positions. The chain file is an annotation file that links the positions between the genome builds. You can download this file from the (UCSC golden path download site)[<a href="https://hgdownload.cse.ucsc.edu/goldenpath/hg19/liftOver/" class="external-link uri">https://hgdownload.cse.ucsc.edu/goldenpath/hg19/liftOver/</a>]. Be sure to download the file that maps between the appropriate builds. We’ll be mapping from hg19 to hg38. We’ve included the chain used in this analysis as a part of the regionalpcs package, which can be accessed in the “extdata” folder as shown in the code below.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert hg19 positions into a GenomicRanges object</span></span>
<span><span class="va">hg19_pos</span> <span class="op">&lt;-</span> <span class="va">new_cpg_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="st">"pos"</span>, <span class="st">"strand"</span>, <span class="st">"probe"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>start <span class="op">=</span> <span class="va">pos</span>, end <span class="op">=</span> <span class="va">pos</span> <span class="op">+</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hg19_pos_gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html" class="external-link">makeGRangesFromDataFrame</a></span><span class="op">(</span><span class="va">hg19_pos</span>, keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Load chain file and liftOver positions</span></span>
<span><span class="va">chain_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"hg19ToHg38.over.chain"</span>,</span>
<span>    package <span class="op">=</span> <span class="st">"regionalpcs"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Using chain file for liftOver"</span>, <span class="va">chain_file</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "Using chain file for liftOver C:/Users/tyeul/AppData/Local/R/win-library/4.3/regionalpcs/extdata/hg19ToHg38.over.chain"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="va">chain_file</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span>
<span><span class="va">chain</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/Chain-class.html" class="external-link">import.chain</a></span><span class="op">(</span><span class="va">chain_file</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hg38_pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/rtracklayer/man/liftOver.html" class="external-link">liftOver</a></span><span class="op">(</span><span class="va">hg19_pos_gr</span>, <span class="va">chain</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Merge the lifted positions back to the original data frame</span></span>
<span><span class="va">formatted_hg38</span> <span class="op">&lt;-</span> <span class="va">hg38_pos</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span>chrom_hg38 <span class="op">=</span> <span class="va">seqnames</span>, pos_hg38 <span class="op">=</span> <span class="va">start</span>, <span class="va">probe</span><span class="op">)</span></span>
<span></span>
<span><span class="va">lifted_cpg_df</span> <span class="op">&lt;-</span> <span class="va">new_cpg_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">formatted_hg38</span>, by <span class="op">=</span> <span class="st">"probe"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">lifted_cpg_df</span><span class="op">)</span></span>
<span><span class="co">#&gt;                    cpg_pos      probe   chr       pos strand chrom_hg38</span></span>
<span><span class="co">#&gt; 1  chr16_53434200_53434201 cg00000029 chr16  53468112      +      chr16</span></span>
<span><span class="co">#&gt; 2  chr15_22838620_22838621 cg00000622 chr15  23034447      +      chr15</span></span>
<span><span class="co">#&gt; 3 chr1_166989202_166989203 cg00001349  chr1 166958439      -       chr1</span></span>
<span><span class="co">#&gt; 4 chr8_119416178_119416179 cg00002464  chr8 120428418      +       chr8</span></span>
<span><span class="co">#&gt; 5 chr6_169751536_169751537 cg00005543  chr6 170151632      +       chr6</span></span>
<span><span class="co">#&gt; 6  chr12_52069532_52069533 cg00006122 chr12  52463316      +      chr12</span></span>
<span><span class="co">#&gt;    pos_hg38</span></span>
<span><span class="co">#&gt; 1  53434200</span></span>
<span><span class="co">#&gt; 2  22838620</span></span>
<span><span class="co">#&gt; 3 166989202</span></span>
<span><span class="co">#&gt; 4 119416178</span></span>
<span><span class="co">#&gt; 5 169751536</span></span>
<span><span class="co">#&gt; 6  52069532</span></span></code></pre></div>
<p>Now that we have accurate genomic positions for each probe and have harmonized genome builds, we can proceed with preprocessing the methylation data.</p>
</div>
</div>
<div class="section level2">
<h2 id="processing-and-filtering-methylation-data">Processing and Filtering Methylation Data<a class="anchor" aria-label="anchor" href="#processing-and-filtering-methylation-data"></a>
</h2>
<div class="section level3">
<h3 id="introduction-1">Introduction<a class="anchor" aria-label="anchor" href="#introduction-1"></a>
</h3>
<p>Before conducting downstream analyses, it is essential to preprocess and clean the methylation data. In this section, we’ll walk you through the steps to remove low variance CpGs and normalize the methylation beta values.</p>
</div>
<div class="section level3">
<h3 id="remove-low-variance-cpgs">Remove Low Variance CpGs<a class="anchor" aria-label="anchor" href="#remove-low-variance-cpgs"></a>
</h3>
<p>Firstly, we aim to filter out low variance CpGs. Variability is a crucial factor, as low variance CpGs may not provide much information for downstream analyses.</p>
<p>In this section, we’ll remove low variance CpGs and normalize our methylation beta values using the inverse normal transform.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Remove CpGs with zero variance</span></span>
<span><span class="va">var_betas</span> <span class="op">&lt;-</span> <span class="va">betas</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">betas</span>, <span class="fl">1</span>, <span class="va">var</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">!=</span> <span class="fl">0</span>, <span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/stats/na.fail.html" class="external-link">na.omit</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">var_betas</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 293 300</span></span></code></pre></div>
<p>We only remove CpGs that have zero variance in this example. You can adjust this threshold according to the requirements of your specific analysis.</p>
</div>
<div class="section level3">
<h3 id="normalize-methylation-values">Normalize Methylation Values<a class="anchor" aria-label="anchor" href="#normalize-methylation-values"></a>
</h3>
<p>Methylation data often exhibit heteroscedasticity. Therefore, we’ll normalize the beta values using inverse normal transformation. For this, we’ll use the <code>RankNorm</code> function from the <code>RNOmni</code> package.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Apply inverse normal transformation to methylation beta values</span></span>
<span><span class="va">int_meth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">var_betas</span>, <span class="fl">1</span>, <span class="va">RankNorm</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>After these preprocessing steps, you will have a dataset ready for downstream analysis with the <code>regionalpcs</code> package. We’ll cover how to perform these analyses in subsequent sections of this tutorial.</p>
</div>
</div>
<div class="section level2">
<h2 id="summarizing-gene-region-types">Summarizing Gene Region Types<a class="anchor" aria-label="anchor" href="#summarizing-gene-region-types"></a>
</h2>
<div class="section level3">
<h3 id="introduction-2">Introduction<a class="anchor" aria-label="anchor" href="#introduction-2"></a>
</h3>
<p>Gene regions are significant functional units of the genome, such as promoters, gene bodies, and intergenic regions. We’ll focus on summarizing these regions to prepare for downstream analyses. We will use the <code>regionalpcs</code> package to perform these tasks.</p>
</div>
<div class="section level3">
<h3 id="load-gene-region-annotations">Load Gene Region Annotations<a class="anchor" aria-label="anchor" href="#load-gene-region-annotations"></a>
</h3>
<p>First, let’s load the gene region annotations. Make sure to align the genomic builds of your annotations and methylation data.</p>
<p><strong>All annotations included with the <code>regionalpcs</code> package are in build hg38.</strong></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the gene region annotation file</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"gene_annots"</span>, package <span class="op">=</span> <span class="st">"regionalpcs"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gene_annots</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 16</span></span>
<span><span class="co">#&gt;   seqnames   start     end width strand tx_id             type   gencode_gene_id</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;      &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;             &lt;chr&gt;  &lt;chr&gt;          </span></span>
<span><span class="co">#&gt; 1 chr1      922928  923927  1000 +      ENST00000420190.6 hg38_… ENSG0000018763…</span></span>
<span><span class="co">#&gt; 2 chr1      959584  960583  1000 +      ENST00000338591.8 hg38_… ENSG0000018796…</span></span>
<span><span class="co">#&gt; 3 chr1      965482  966481  1000 +      ENST00000379410.8 hg38_… ENSG0000018758…</span></span>
<span><span class="co">#&gt; 4 chr1     1000138 1001137  1000 +      ENST00000624697.4 hg38_… ENSG0000018760…</span></span>
<span><span class="co">#&gt; 5 chr1     1019120 1020119  1000 +      ENST00000379370.7 hg38_… ENSG0000018815…</span></span>
<span><span class="co">#&gt; 6 chr1     1172903 1173902  1000 +      ENST00000379290.5 hg38_… ENSG0000016257…</span></span>
<span><span class="co">#&gt; # ℹ 8 more variables: gencode_gene_type &lt;chr&gt;, gencode_gene_name &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   transcript_type &lt;chr&gt;, transcript_name &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   transcript_support_level &lt;dbl&gt;, tag &lt;chr&gt;, is_canonical &lt;lgl&gt;,</span></span>
<span><span class="co">#&gt; #   gencode_region &lt;chr&gt;</span></span></code></pre></div>
<p>The <code>gene_annots</code> dataset includes annotations for various gene regions.</p>
</div>
<div class="section level3">
<h3 id="create-a-region-map">Create a Region Map<a class="anchor" aria-label="anchor" href="#create-a-region-map"></a>
</h3>
<p>Before summarizing gene regions using <code>compute_regional_pcs</code>, we need to create a region map that assigns CpGs to gene regions. This map enables us to identify which CpGs fall into each gene region.</p>
<div class="section level4">
<h4 id="extract-cpg-positions">Extract CpG Positions<a class="anchor" aria-label="anchor" href="#extract-cpg-positions"></a>
</h4>
<p>Start by extracting the CpG positions from your methylation data frame’s row names.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">int_meth</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="co">#&gt;                                     TCGA-EJ-7781-11A TCGA-BH-A1FE-11B</span></span>
<span><span class="co">#&gt; chr16_53434200_53434201_cg00000029       -0.38948253       -0.9465265</span></span>
<span><span class="co">#&gt; chr15_22838620_22838621_cg00000622       -0.03757696       -1.1171114</span></span>
<span><span class="co">#&gt; chr1_166989202_166989203_cg00001349      -0.87083034       -0.6274087</span></span>
<span><span class="co">#&gt; chr8_119416178_119416179_cg00002464       0.13818831        0.1045460</span></span>
<span><span class="co">#&gt; chr6_169751536_169751537_cg00005543       0.38049184        0.2488238</span></span>
<span><span class="co">#&gt; chr12_52069532_52069533_cg00006122        0.27474294        0.6274087</span></span>
<span><span class="co">#&gt;                                     TCGA-BH-A0C3-11A TCGA-E9-A1N6-11A</span></span>
<span><span class="co">#&gt; chr16_53434200_53434201_cg00000029        -1.1171114       -1.4370361</span></span>
<span><span class="co">#&gt; chr15_22838620_22838621_cg00000622         1.2160129       -0.6376059</span></span>
<span><span class="co">#&gt; chr1_166989202_166989203_cg00001349       -0.3894825        0.3894825</span></span>
<span><span class="co">#&gt; chr8_119416178_119416179_cg00002464        0.2402219       -0.2574441</span></span>
<span><span class="co">#&gt; chr6_169751536_169751537_cg00005543        0.9080280       -1.5657713</span></span>
<span><span class="co">#&gt; chr12_52069532_52069533_cg00006122         0.3271598       -0.1129440</span></span>
<span><span class="co"># Extract CpG information</span></span>
<span><span class="va">cpg_info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>cpg_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">int_meth</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span><span class="va">cpg_id</span>,</span>
<span>        into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chrom"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"cpg_name"</span><span class="op">)</span>,</span>
<span>        sep <span class="op">=</span> <span class="st">"_"</span>, remove <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>    <span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">cpg_info</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                cpg_id chrom     start       end   cpg_name</span></span>
<span><span class="co">#&gt; 1  chr16_53434200_53434201_cg00000029 chr16  53434200  53434201 cg00000029</span></span>
<span><span class="co">#&gt; 2  chr15_22838620_22838621_cg00000622 chr15  22838620  22838621 cg00000622</span></span>
<span><span class="co">#&gt; 3 chr1_166989202_166989203_cg00001349  chr1 166989202 166989203 cg00001349</span></span>
<span><span class="co">#&gt; 4 chr8_119416178_119416179_cg00002464  chr8 119416178 119416179 cg00002464</span></span>
<span><span class="co">#&gt; 5 chr6_169751536_169751537_cg00005543  chr6 169751536 169751537 cg00005543</span></span>
<span><span class="co">#&gt; 6  chr12_52069532_52069533_cg00006122 chr12  52069532  52069533 cg00006122</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="convert-to-genomicranges-and-find-overlaps">Convert to GenomicRanges and Find Overlaps<a class="anchor" aria-label="anchor" href="#convert-to-genomicranges-and-find-overlaps"></a>
</h4>
<p>Next, we’ll use the <code>GenomicRanges</code> package to find overlaps between CpGs and gene regions.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Convert to GenomicRanges</span></span>
<span><span class="va">cpg_gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html" class="external-link">makeGRangesFromDataFrame</a></span><span class="op">(</span><span class="va">cpg_info</span>, keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">annots_gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/makeGRangesFromDataFrame.html" class="external-link">makeGRangesFromDataFrame</a></span><span class="op">(</span><span class="va">gene_annots</span>, keep.extra.columns <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Find overlaps between the two GRanges objects</span></span>
<span><span class="va">overlaps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/findOverlaps-methods.html" class="external-link">findOverlaps</a></span><span class="op">(</span>query <span class="op">=</span> <span class="va">cpg_gr</span>, subject <span class="op">=</span> <span class="va">annots_gr</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">overlaps</span><span class="op">)</span></span>
<span><span class="co">#&gt;   queryHits subjectHits</span></span>
<span><span class="co">#&gt; 1         1       12678</span></span>
<span><span class="co">#&gt; 2         2       11904</span></span>
<span><span class="co">#&gt; 3         3         679</span></span>
<span><span class="co">#&gt; 4         4        7360</span></span>
<span><span class="co">#&gt; 5         5        5877</span></span>
<span><span class="co">#&gt; 6         6       10306</span></span>
<span></span>
<span><span class="co"># Match overlaps</span></span>
<span><span class="va">matched_cpg</span> <span class="op">&lt;-</span> <span class="va">cpg_gr</span><span class="op">[</span><span class="va">overlaps</span><span class="op">$</span><span class="va">queryHits</span>, <span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">cpg_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Select overlapped rows and just keep the columns we need</span></span>
<span><span class="va">matched_annots</span> <span class="op">&lt;-</span> <span class="va">annots_gr</span><span class="op">[</span><span class="va">overlaps</span><span class="op">$</span><span class="va">subjectHits</span>, <span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">gencode_gene_id</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Combine the matched CpGs and gene annotations to form the region map</span></span>
<span><span class="va">region_map</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">matched_annots</span>, <span class="va">matched_cpg</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">region_map</span><span class="op">)</span></span>
<span><span class="co">#&gt;      gencode_gene_id                              cpg_id</span></span>
<span><span class="co">#&gt; 1 ENSG00000103479.16  chr16_53434200_53434201_cg00000029</span></span>
<span><span class="co">#&gt; 2 ENSG00000140157.14  chr15_22838620_22838621_cg00000622</span></span>
<span><span class="co">#&gt; 3 ENSG00000143194.13 chr1_166989202_166989203_cg00001349</span></span>
<span><span class="co">#&gt; 4  ENSG00000136999.5 chr8_119416178_119416179_cg00002464</span></span>
<span><span class="co">#&gt; 5 ENSG00000130023.16 chr6_169751536_169751537_cg00005543</span></span>
<span><span class="co">#&gt; 6 ENSG00000123395.14  chr12_52069532_52069533_cg00006122</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">region_map</span><span class="op">$</span><span class="va">gencode_gene_id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 52</span></span></code></pre></div>
<p>With these steps, you’ll have a region map that assigns CpGs to specific gene regions, which can be essential for downstream analyses.</p>
<p><a name="summarizing-gene-regions"></a></p>
</div>
</div>
<div class="section level3">
<h3 id="summarizing-gene-regions-with-regional-principal-components">Summarizing Gene Regions with Regional Principal Components<a class="anchor" aria-label="anchor" href="#summarizing-gene-regions-with-regional-principal-components"></a>
</h3>
<p>In this final section, we’ll summarize gene regions using Principal Components (PCs) to capture the maximum variation. We’ll utilize the <code>compute_regional_pcs</code> function from the <code>regionalpcs</code> package for this.</p>
<div class="section level4">
<h4 id="compute-regional-pcs">Compute Regional PCs<a class="anchor" aria-label="anchor" href="#compute-regional-pcs"></a>
</h4>
<p>Let’s calculate the regional PCs using a subset of our gene regions for demonstration purposes.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Display head of region map</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">region_map</span><span class="op">)</span></span>
<span><span class="co">#&gt;      gencode_gene_id                              cpg_id</span></span>
<span><span class="co">#&gt; 1 ENSG00000103479.16  chr16_53434200_53434201_cg00000029</span></span>
<span><span class="co">#&gt; 2 ENSG00000140157.14  chr15_22838620_22838621_cg00000622</span></span>
<span><span class="co">#&gt; 3 ENSG00000143194.13 chr1_166989202_166989203_cg00001349</span></span>
<span><span class="co">#&gt; 4  ENSG00000136999.5 chr8_119416178_119416179_cg00002464</span></span>
<span><span class="co">#&gt; 5 ENSG00000130023.16 chr6_169751536_169751537_cg00005543</span></span>
<span><span class="co">#&gt; 6 ENSG00000123395.14  chr12_52069532_52069533_cg00006122</span></span>
<span></span>
<span><span class="co"># Subset the region map</span></span>
<span><span class="va">sub_region_map</span> <span class="op">&lt;-</span> <span class="va">region_map</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">gencode_gene_id</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">region_map</span><span class="op">$</span><span class="va">gencode_gene_id</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">1000</span><span class="op">]</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Compute regional PCs</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/compute_regional_pcs.html">compute_regional_pcs</a></span><span class="op">(</span><span class="va">int_meth</span>, <span class="va">sub_region_map</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using Gavish-Donoho method</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="inspecting-the-output">Inspecting the Output<a class="anchor" aria-label="anchor" href="#inspecting-the-output"></a>
</h4>
<p>The function returns a list containing multiple elements. Let’s first look at what these elements are.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Inspect the output list elements</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "regional_pcs"     "percent_variance" "loadings"         "info"</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="extracting-and-viewing-regional-pcs">Extracting and Viewing Regional PCs<a class="anchor" aria-label="anchor" href="#extracting-and-viewing-regional-pcs"></a>
</h4>
<p>The first element (<code>res$regional_pcs</code>) is a data frame containing the calculated regional PCs.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract regional PCs</span></span>
<span><span class="va">regional_pcs</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="va">regional_pcs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">regional_pcs</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="co">#&gt;                        TCGA-EJ-7781-11A TCGA-BH-A1FE-11B TCGA-BH-A0C3-11A</span></span>
<span><span class="co">#&gt; ENSG00000103479.16-PC1       -0.8210157      -1.60659103       -1.0541882</span></span>
<span><span class="co">#&gt; ENSG00000140157.14-PC1       -0.1392372      -1.47241611        1.3317941</span></span>
<span><span class="co">#&gt; ENSG00000143194.13-PC1       -1.7895014      -3.07334567       -3.1270943</span></span>
<span><span class="co">#&gt; ENSG00000136999.5-PC1         0.4498687       0.11381324       -0.3555689</span></span>
<span><span class="co">#&gt; ENSG00000130023.16-PC1       -0.1838917       0.05055839       -0.8546051</span></span>
<span><span class="co">#&gt; ENSG00000123395.14-PC1       -1.1961240       0.36432710        0.7234163</span></span>
<span><span class="co">#&gt;                        TCGA-E9-A1N6-11A</span></span>
<span><span class="co">#&gt; ENSG00000103479.16-PC1       -1.6169547</span></span>
<span><span class="co">#&gt; ENSG00000140157.14-PC1       -0.6519472</span></span>
<span><span class="co">#&gt; ENSG00000143194.13-PC1       -1.0122026</span></span>
<span><span class="co">#&gt; ENSG00000136999.5-PC1         1.2443734</span></span>
<span><span class="co">#&gt; ENSG00000130023.16-PC1        2.1260186</span></span>
<span><span class="co">#&gt; ENSG00000123395.14-PC1        1.3592205</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="understanding-the-results">Understanding the Results<a class="anchor" aria-label="anchor" href="#understanding-the-results"></a>
</h4>
<p>The output is a data frame with regional PCs for each region as rows and our samples as columns. This is our new representation of methylation values, now on a gene regional PC scale. We can feed these into downstream analyses as is.</p>
<p>The number of regional PCs representing each gene region was determined by the Gavish-Donoho method. This method allows us to identify PCs that capture actual signal in our data and not the noise that is inherent in any dataset. To explore alternative methods, we can change the <code>pc_method</code> parameter.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Count the number of unique gene regions and PCs</span></span>
<span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>gene_pc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">regional_pcs</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span><span class="va">gene_pc</span>, into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"pc"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"-"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">regions</span><span class="op">)</span></span>
<span><span class="co">#&gt;                 gene  pc</span></span>
<span><span class="co">#&gt; 1 ENSG00000103479.16 PC1</span></span>
<span><span class="co">#&gt; 2 ENSG00000140157.14 PC1</span></span>
<span><span class="co">#&gt; 3 ENSG00000143194.13 PC1</span></span>
<span><span class="co">#&gt; 4  ENSG00000136999.5 PC1</span></span>
<span><span class="co">#&gt; 5 ENSG00000130023.16 PC1</span></span>
<span><span class="co">#&gt; 6 ENSG00000123395.14 PC1</span></span>
<span></span>
<span><span class="co"># number of genes that have been summarized</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">regions</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 52</span></span>
<span></span>
<span><span class="co"># how many of each PC did we get</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">regions</span><span class="op">$</span><span class="va">pc</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; PC1 </span></span>
<span><span class="co">#&gt;  52</span></span></code></pre></div>
<p>We have summarized each of our genes using just one PC. The number of PCs depends on three main factors: the number of samples, the number of CpGs in the gene region, and the noise in the methylation data.</p>
<p>By default, the <code>compute_regional_pcs</code> function uses the Gavish-Donoho method. However, we can also use the Marcenko-Pasteur method by setting the <code>pc_method</code> parameter:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Using Marcenko-Pasteur method</span></span>
<span><span class="va">mp_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/compute_regional_pcs.html">compute_regional_pcs</a></span><span class="op">(</span><span class="va">int_meth</span>, <span class="va">sub_region_map</span>, pc_method <span class="op">=</span> <span class="st">"mp"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using Marchenko-Pastur method</span></span>
<span></span>
<span><span class="co"># select the regional pcs</span></span>
<span><span class="va">mp_regional_pcs</span> <span class="op">&lt;-</span> <span class="va">mp_res</span><span class="op">$</span><span class="va">regional_pcs</span></span>
<span></span>
<span><span class="co"># separate the genes from the pc numbers</span></span>
<span><span class="va">mp_regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>gene_pc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">mp_regional_pcs</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span><span class="va">gene_pc</span>, into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"pc"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"-"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mp_regions</span><span class="op">)</span></span>
<span><span class="co">#&gt;                 gene  pc</span></span>
<span><span class="co">#&gt; 1 ENSG00000103479.16 PC1</span></span>
<span><span class="co">#&gt; 2 ENSG00000103479.16 PC2</span></span>
<span><span class="co">#&gt; 3 ENSG00000140157.14 PC1</span></span>
<span><span class="co">#&gt; 4 ENSG00000140157.14 PC2</span></span>
<span><span class="co">#&gt; 5 ENSG00000143194.13 PC1</span></span>
<span><span class="co">#&gt; 6 ENSG00000143194.13 PC2</span></span>
<span></span>
<span><span class="co"># number of genes that have been summarized</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">mp_regions</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 52</span></span>
<span></span>
<span><span class="co"># how many of each PC did we get</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">mp_regions</span><span class="op">$</span><span class="va">pc</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; PC1 PC2 </span></span>
<span><span class="co">#&gt;  52  26</span></span></code></pre></div>
<p>The Marcenko-Pasteur and the Gavish-Donoho methods are both based on Random Matrix Theory, and they aim to identify the number of significant PCs that capture the true signal in the data and not just the noise. However, these methods differ in how they select the number of significant PCs. The Marcenko-Pasteur method typically selects more PCs to represent a gene region compared to the Gavish-Donoho method. This may be due to the different ways in which the two methods estimate the noise level in the data.</p>
<p>Ultimately, the choice between the two methods depends on the specific needs and goals of the analysis. The Gavish-Donoho method tends to provide more conservative results, while the Marcenko-Pasteur method may capture more of the underlying signal in the data. Researchers should carefully consider their objectives and the characteristics of their data when selecting a method.</p>
</div>
</div>
</div>
</div>
<div class="section level1">
<h1 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h1>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Windows 10 x64 (build 19045)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=English_United States.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=English_United States.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=English_United States.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                          </span></span>
<span><span class="co">#&gt; [5] LC_TIME=English_United States.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/Los_Angeles</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">#&gt; [8] methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] dplyr_1.1.2                                       </span></span>
<span><span class="co">#&gt;  [2] tibble_3.2.1                                      </span></span>
<span><span class="co">#&gt;  [3] tidyr_1.3.0                                       </span></span>
<span><span class="co">#&gt;  [4] magrittr_2.0.3                                    </span></span>
<span><span class="co">#&gt;  [5] liftOver_1.25.0                                   </span></span>
<span><span class="co">#&gt;  [6] Homo.sapiens_1.3.1                                </span></span>
<span><span class="co">#&gt;  [7] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2           </span></span>
<span><span class="co">#&gt;  [8] org.Hs.eg.db_3.17.0                               </span></span>
<span><span class="co">#&gt;  [9] GO.db_3.17.0                                      </span></span>
<span><span class="co">#&gt; [10] OrganismDbi_1.43.0                                </span></span>
<span><span class="co">#&gt; [11] GenomicFeatures_1.53.1                            </span></span>
<span><span class="co">#&gt; [12] AnnotationDbi_1.63.2                              </span></span>
<span><span class="co">#&gt; [13] rtracklayer_1.61.1                                </span></span>
<span><span class="co">#&gt; [14] gwascat_2.33.0                                    </span></span>
<span><span class="co">#&gt; [15] IlluminaHumanMethylation450kanno.ilmn12.hg19_0.6.1</span></span>
<span><span class="co">#&gt; [16] minfi_1.47.0                                      </span></span>
<span><span class="co">#&gt; [17] bumphunter_1.43.0                                 </span></span>
<span><span class="co">#&gt; [18] locfit_1.5-9.8                                    </span></span>
<span><span class="co">#&gt; [19] iterators_1.0.14                                  </span></span>
<span><span class="co">#&gt; [20] foreach_1.5.2                                     </span></span>
<span><span class="co">#&gt; [21] Biostrings_2.69.2                                 </span></span>
<span><span class="co">#&gt; [22] XVector_0.41.1                                    </span></span>
<span><span class="co">#&gt; [23] SummarizedExperiment_1.31.1                       </span></span>
<span><span class="co">#&gt; [24] Biobase_2.61.0                                    </span></span>
<span><span class="co">#&gt; [25] MatrixGenerics_1.13.1                             </span></span>
<span><span class="co">#&gt; [26] matrixStats_1.0.0                                 </span></span>
<span><span class="co">#&gt; [27] GenomicRanges_1.53.1                              </span></span>
<span><span class="co">#&gt; [28] GenomeInfoDb_1.37.2                               </span></span>
<span><span class="co">#&gt; [29] IRanges_2.35.2                                    </span></span>
<span><span class="co">#&gt; [30] S4Vectors_0.39.1                                  </span></span>
<span><span class="co">#&gt; [31] BiocGenerics_0.47.0                               </span></span>
<span><span class="co">#&gt; [32] RNOmni_1.0.1                                      </span></span>
<span><span class="co">#&gt; [33] regionalpcs_0.99.1                                </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] splines_4.3.1             BiocIO_1.11.0            </span></span>
<span><span class="co">#&gt;   [3] bitops_1.0-7              filelock_1.0.2           </span></span>
<span><span class="co">#&gt;   [5] preprocessCore_1.63.1     graph_1.79.0             </span></span>
<span><span class="co">#&gt;   [7] XML_3.99-0.14             lifecycle_1.0.3          </span></span>
<span><span class="co">#&gt;   [9] lattice_0.21-8            MASS_7.3-60              </span></span>
<span><span class="co">#&gt;  [11] base64_2.0.1              scrime_1.3.5             </span></span>
<span><span class="co">#&gt;  [13] limma_3.57.7              rmarkdown_2.24           </span></span>
<span><span class="co">#&gt;  [15] yaml_2.3.7                doRNG_1.8.6              </span></span>
<span><span class="co">#&gt;  [17] askpass_1.1               cowplot_1.1.1            </span></span>
<span><span class="co">#&gt;  [19] DBI_1.1.3                 RColorBrewer_1.1-3       </span></span>
<span><span class="co">#&gt;  [21] abind_1.4-5               zlibbioc_1.47.0          </span></span>
<span><span class="co">#&gt;  [23] quadprog_1.5-8            purrr_1.0.2              </span></span>
<span><span class="co">#&gt;  [25] RCurl_1.98-1.12           VariantAnnotation_1.47.1 </span></span>
<span><span class="co">#&gt;  [27] rappdirs_0.3.3            GenomeInfoDbData_1.2.10  </span></span>
<span><span class="co">#&gt;  [29] RMTstat_0.3.1             ggrepel_0.9.3            </span></span>
<span><span class="co">#&gt;  [31] irlba_2.3.5.1             genefilter_1.83.1        </span></span>
<span><span class="co">#&gt;  [33] dqrng_0.3.0               annotate_1.79.0          </span></span>
<span><span class="co">#&gt;  [35] DelayedMatrixStats_1.23.4 codetools_0.2-19         </span></span>
<span><span class="co">#&gt;  [37] DelayedArray_0.27.10      xml2_1.3.5               </span></span>
<span><span class="co">#&gt;  [39] tidyselect_1.2.0          ScaledMatrix_1.9.1       </span></span>
<span><span class="co">#&gt;  [41] beanplot_1.3.1            BiocFileCache_2.9.1      </span></span>
<span><span class="co">#&gt;  [43] illuminaio_0.43.0         GenomicAlignments_1.37.0 </span></span>
<span><span class="co">#&gt;  [45] multtest_2.57.0           survival_3.5-7           </span></span>
<span><span class="co">#&gt;  [47] tools_4.3.1               progress_1.2.2           </span></span>
<span><span class="co">#&gt;  [49] Rcpp_1.0.11               glue_1.6.2               </span></span>
<span><span class="co">#&gt;  [51] SparseArray_1.1.11        xfun_0.40                </span></span>
<span><span class="co">#&gt;  [53] HDF5Array_1.29.3          withr_2.5.0              </span></span>
<span><span class="co">#&gt;  [55] BiocManager_1.30.22       fastmap_1.1.1            </span></span>
<span><span class="co">#&gt;  [57] rhdf5filters_1.13.5       fansi_1.0.4              </span></span>
<span><span class="co">#&gt;  [59] openssl_2.1.0             rsvd_1.0.5               </span></span>
<span><span class="co">#&gt;  [61] digest_0.6.33             R6_2.5.1                 </span></span>
<span><span class="co">#&gt;  [63] colorspace_2.1-0          biomaRt_2.57.1           </span></span>
<span><span class="co">#&gt;  [65] RSQLite_2.3.1             utf8_1.2.3               </span></span>
<span><span class="co">#&gt;  [67] generics_0.1.3            data.table_1.14.8        </span></span>
<span><span class="co">#&gt;  [69] prettyunits_1.1.1         httr_1.4.7               </span></span>
<span><span class="co">#&gt;  [71] S4Arrays_1.1.5            pkgconfig_2.0.3          </span></span>
<span><span class="co">#&gt;  [73] gtable_0.3.3              blob_1.2.4               </span></span>
<span><span class="co">#&gt;  [75] siggenes_1.75.0           htmltools_0.5.6          </span></span>
<span><span class="co">#&gt;  [77] RBGL_1.77.1               scales_1.2.1             </span></span>
<span><span class="co">#&gt;  [79] png_0.1-8                 knitr_1.43               </span></span>
<span><span class="co">#&gt;  [81] rstudioapi_0.15.0         reshape2_1.4.4           </span></span>
<span><span class="co">#&gt;  [83] tzdb_0.4.0                rjson_0.2.21             </span></span>
<span><span class="co">#&gt;  [85] nlme_3.1-163              curl_5.0.2               </span></span>
<span><span class="co">#&gt;  [87] cachem_1.0.8              rhdf5_2.45.1             </span></span>
<span><span class="co">#&gt;  [89] stringr_1.5.0             restfulr_0.0.15          </span></span>
<span><span class="co">#&gt;  [91] GEOquery_2.69.0           pillar_1.9.0             </span></span>
<span><span class="co">#&gt;  [93] grid_4.3.1                reshape_0.8.9            </span></span>
<span><span class="co">#&gt;  [95] vctrs_0.6.3               BiocSingular_1.17.1      </span></span>
<span><span class="co">#&gt;  [97] beachmat_2.17.15          dbplyr_2.3.3             </span></span>
<span><span class="co">#&gt;  [99] xtable_1.8-4              evaluate_0.21            </span></span>
<span><span class="co">#&gt; [101] readr_2.1.4               cli_3.6.1                </span></span>
<span><span class="co">#&gt; [103] compiler_4.3.1            Rsamtools_2.17.0         </span></span>
<span><span class="co">#&gt; [105] rlang_1.1.1               crayon_1.5.2             </span></span>
<span><span class="co">#&gt; [107] rngtools_1.5.2            nor1mix_1.3-0            </span></span>
<span><span class="co">#&gt; [109] mclust_6.0.0              plyr_1.8.8               </span></span>
<span><span class="co">#&gt; [111] stringi_1.7.12            BiocParallel_1.35.3      </span></span>
<span><span class="co">#&gt; [113] munsell_0.5.0             PCAtools_2.13.0          </span></span>
<span><span class="co">#&gt; [115] Matrix_1.6-1              BSgenome_1.69.0          </span></span>
<span><span class="co">#&gt; [117] hms_1.1.3                 sparseMatrixStats_1.13.4 </span></span>
<span><span class="co">#&gt; [119] bit64_4.0.5               ggplot2_3.4.3            </span></span>
<span><span class="co">#&gt; [121] Rhdf5lib_1.23.0           KEGGREST_1.41.0          </span></span>
<span><span class="co">#&gt; [123] statmod_1.5.0             memoise_2.0.1            </span></span>
<span><span class="co">#&gt; [125] snpStats_1.51.0           bit_4.0.5</span></span></code></pre></div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/regionalpcs" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/tyeulalio/regionalpcs/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/tyeulalio/regionalpcs/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing regionalpcs</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Tiffany Eulalio <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-7084-9646" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Tiffany Eulalio.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
