<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="regionalpcs">
<title>regionalpcs â€¢ regionalpcs</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="regionalpcs">
<meta property="og:description" content="regionalpcs">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">regionalpcs</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.4</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/regionalpcs.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/tyeulalio/regionalpcs/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>regionalpcs</h1>
                        <h4 data-toc-skip class="author">Tiffany
Eulalio</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/tyeulalio/regionalpcs/blob/HEAD/vignettes/regionalpcs.Rmd" class="external-link"><code>vignettes/regionalpcs.Rmd</code></a></small>
      <div class="d-none name"><code>regionalpcs.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="regionalpcs">regionalpcs<a class="anchor" aria-label="anchor" href="#regionalpcs"></a>
</h2>
<p>Navigating the complexity of DNA methylation data poses substantial
challenges due to its intricate biological patterns. The
<code>regionalpcs</code> package is conceived to address the substantial
need for enhanced summarization and interpretation at a regional level.
Traditional methodologies, while foundational, may not fully encapsulate
the biological nuances of methylation patterns, thereby potentially
yielding interpretations that may be suboptimal or veer towards
inaccuracies. This package introduces and utilizes regional principal
components (rPCs), designed to adeptly capture biologically relevant
signals embedded within DNA methylation data. Through the implementation
of rPCs, researchers can gain new insights into complex interactions and
effects in methylation data that might otherwise be missed.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The <code>regionalpcs</code> package can be easily installed from
Bioconductor, providing you with the latest stable version suitable for
general use. Alternatively, for those interested in exploring or
utilizing the latest features and developments, the GitHub version can
be installed directly.</p>
<div class="section level4">
<h4 id="bioconductor-installation">Bioconductor Installation<a class="anchor" aria-label="anchor" href="#bioconductor-installation"></a>
</h4>
<p>Install <code>regionalpcs</code> from Bioconductor using the command
below. Ensure that your R version is compatible with the package version
available on Bioconductor for smooth installation and functionality.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"regionalpcs"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="development-version-installation">Development Version Installation<a class="anchor" aria-label="anchor" href="#development-version-installation"></a>
</h4>
<p>To access the development version of <code>regionalpcs</code>
directly from GitHub, which might include new features or enhancements
not yet available in the Bioconductor version, use the following
commands. Note that the development version might be less stable than
the officially released version.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install devtools package if not already installed</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># download and install the regionalpcs package</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"tyeulalio/regionalpcs"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="regionalpcs-r-package-tutorial">
<code>regionalpcs</code> R Package Tutorial<a class="anchor" aria-label="anchor" href="#regionalpcs-r-package-tutorial"></a>
</h2>
<div class="section level3">
<h3 id="loading-required-packages">Loading Required Packages<a class="anchor" aria-label="anchor" href="#loading-required-packages"></a>
</h3>
<p>This tutorial depends on several Bioconductor packages. These
packages should be loaded at the beginning of the analysis.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/tyeulalio/regionalpcs" class="external-link">regionalpcs</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">minfiData</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span><span class="op">)</span></span></code></pre></div>
<p>Letâ€™s proceed to load the packages, briefly understanding their roles
in this tutorial.</p>
<ul>
<li><p><code>regionalpcs</code>: Primary package for summarizing and
interpreting DNA methylation data at a regional level.</p></li>
<li><p><code>GenomicRanges</code>: Facilitates representation and
manipulation of genomic intervals and variables defined along a
genome.</p></li>
<li><p><code>tidyr</code>, <code>tibble</code>, <code>dplyr</code>:
Assist in data tidying, representation, and manipulation.</p></li>
<li><p><code>minfiData</code>: Provides example Illumina 450k data,
aiding in the demonstration of <code>regionalpcs</code>
functionalities.</p></li>
<li><p><code>TxDb.Hsapiens.UCSC.hg19.knownGene</code>: Accommodates
transcriptome data, useful for annotating results.</p></li>
</ul>
<p>Once the packages are loaded, we can start using the functions
provided by each package.</p>
</div>
<div class="section level3">
<h3 id="load-the-dataset">Load the Dataset<a class="anchor" aria-label="anchor" href="#load-the-dataset"></a>
</h3>
<div class="section level4">
<h4 id="loading-minfi-sample-dataset">Loading Minfi Sample Dataset<a class="anchor" aria-label="anchor" href="#loading-minfi-sample-dataset"></a>
</h4>
<p>In this tutorial, weâ€™ll utilize a sample dataset,
<code>MsetEx.sub</code>, which is a subset derived from Illuminaâ€™s Human
Methylation 450k dataset, specifically preprocessed to contain 600 CpGs
across 6 samples. The dataset is stored in a <code>MethylSet</code>
object, which is commonly used to represent methylation data.</p>
<p>The methylation M-values, denoting the proportion of methylated
cytosines at a particular CpG site, will be extracted from this dataset
for our subsequent analyses.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the MethylSet data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">MsetEx.sub</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Display the first few rows of the dataset for a preliminary view</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">MsetEx.sub</span><span class="op">)</span></span>
<span><span class="co">#&gt; class: MethylSet </span></span>
<span><span class="co">#&gt; dim: 6 6 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(2): Meth Unmeth</span></span>
<span><span class="co">#&gt; rownames(6): cg00050873 cg00212031 ... cg00455876 cg01707559</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(6): 5723646052_R02C02 5723646052_R04C01 ... 5723646053_R05C02</span></span>
<span><span class="co">#&gt;   5723646053_R06C02</span></span>
<span><span class="co">#&gt; colData names(13): Sample_Name Sample_Well ... Basename filenames</span></span>
<span><span class="co">#&gt; Annotation</span></span>
<span><span class="co">#&gt;   array: IlluminaHumanMethylation450k</span></span>
<span><span class="co">#&gt;   annotation: ilmn12.hg19</span></span>
<span><span class="co">#&gt; Preprocessing</span></span>
<span><span class="co">#&gt;   Method: Raw (no normalization or bg correction)</span></span>
<span><span class="co">#&gt;   minfi version: 1.21.2</span></span>
<span><span class="co">#&gt;   Manifest version: 0.4.0</span></span>
<span></span>
<span><span class="co"># Extract methylation M-values from the MethylSet</span></span>
<span><span class="co"># M-values are logit-transformed beta-values and are often used in differential</span></span>
<span><span class="co"># methylation analysis for improved statistical performance.</span></span>
<span><span class="va">mvals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/minfi/man/MethylSet-class.html" class="external-link">getM</a></span><span class="op">(</span><span class="va">MsetEx.sub</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Display the extracted methylation beta values</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mvals</span><span class="op">)</span></span>
<span><span class="co">#&gt;            5723646052_R02C02 5723646052_R04C01 5723646052_R05C02</span></span>
<span><span class="co">#&gt; cg00050873          3.502348         0.4414491          4.340695</span></span>
<span><span class="co">#&gt; cg00212031         -3.273751         0.9234662         -2.614777</span></span>
<span><span class="co">#&gt; cg00213748          2.076816        -0.1309465          1.260995</span></span>
<span><span class="co">#&gt; cg00214611         -3.438838         1.7463950         -2.270551</span></span>
<span><span class="co">#&gt; cg00455876          1.839010        -0.9927320          1.619479</span></span>
<span><span class="co">#&gt; cg01707559         -3.303987        -0.6433201         -3.540887</span></span>
<span><span class="co">#&gt;            5723646053_R04C02 5723646053_R05C02 5723646053_R06C02</span></span>
<span><span class="co">#&gt; cg00050873        0.24458355        -0.3219281         0.2744392</span></span>
<span><span class="co">#&gt; cg00212031       -0.21052257        -0.6861413        -0.1397595</span></span>
<span><span class="co">#&gt; cg00213748       -1.10373279        -1.6616553        -0.1270869</span></span>
<span><span class="co">#&gt; cg00214611        0.29029649        -0.2103599        -0.6138630</span></span>
<span><span class="co">#&gt; cg00455876       -0.09504721        -0.2854655         0.6361273</span></span>
<span><span class="co">#&gt; cg01707559       -0.74835377        -0.4678048        -1.1345421</span></span></code></pre></div>
<p>Note that <code>MsetEx.sub</code> provides a manageable slice of data
that we can utilize to illustrate the capabilities of
<code>regionalpcs</code> without requiring extensive computational
resources.</p>
<p>Now that we have our dataset loaded and methylation values extracted,
letâ€™s proceed with demonstrating the core functionalities of
<code>regionalpcs</code>.</p>
</div>
</div>
<div class="section level3">
<h3 id="obtaining-methylation-array-probe-positions">Obtaining Methylation Array Probe Positions<a class="anchor" aria-label="anchor" href="#obtaining-methylation-array-probe-positions"></a>
</h3>
<div class="section level4">
<h4 id="load-illumina-450k-array-probe-positions">Load Illumina 450k Array Probe Positions<a class="anchor" aria-label="anchor" href="#load-illumina-450k-array-probe-positions"></a>
</h4>
<p>In this step, weâ€™ll obtain the genomic coordinates of the CpG sites
in our methylation dataset using the 450k array probe annotations using
the <code>minfi</code> package.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Map the methylation data to genomic coordinates using the mapToGenome</span></span>
<span><span class="co"># function. This creates a GenomicMethylSet (gset) which includes genomic </span></span>
<span><span class="co"># position information for each probe.</span></span>
<span><span class="va">gset</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/minfi/man/mapToGenome-methods.html" class="external-link">mapToGenome</a></span><span class="op">(</span><span class="va">MsetEx.sub</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Display the GenomicMethylSet object to observe the structure and initial </span></span>
<span><span class="co"># entries.</span></span>
<span><span class="va">gset</span></span>
<span><span class="co">#&gt; class: GenomicMethylSet </span></span>
<span><span class="co">#&gt; dim: 600 6 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(2): Meth Unmeth</span></span>
<span><span class="co">#&gt; rownames(600): cg01003813 cg03723195 ... cg03930849 cg08265308</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(6): 5723646052_R02C02 5723646052_R04C01 ... 5723646053_R05C02</span></span>
<span><span class="co">#&gt;   5723646053_R06C02</span></span>
<span><span class="co">#&gt; colData names(13): Sample_Name Sample_Well ... Basename filenames</span></span>
<span><span class="co">#&gt; Annotation</span></span>
<span><span class="co">#&gt;   array: IlluminaHumanMethylation450k</span></span>
<span><span class="co">#&gt;   annotation: ilmn12.hg19</span></span>
<span><span class="co">#&gt; Preprocessing</span></span>
<span><span class="co">#&gt;   Method: Raw (no normalization or bg correction)</span></span>
<span><span class="co">#&gt;   minfi version: 1.21.2</span></span>
<span><span class="co">#&gt;   Manifest version: 0.4.0</span></span>
<span></span>
<span><span class="co"># Convert the genomic position data into a GRanges object, enabling genomic </span></span>
<span><span class="co"># range operations in subsequent analyses.</span></span>
<span><span class="co"># The GRanges object (cpg_gr) provides a versatile structure for handling </span></span>
<span><span class="co"># genomic coordinates in R/Bioconductor.</span></span>
<span><span class="va">cpg_gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomicRanges/man/genomic-range-squeezers.html" class="external-link">granges</a></span><span class="op">(</span><span class="va">gset</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Display the GRanges object for a preliminary view of the genomic coordinates.</span></span>
<span><span class="va">cpg_gr</span></span>
<span><span class="co">#&gt; GRanges object with 600 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;              seqnames    ranges strand</span></span>
<span><span class="co">#&gt;                 &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   cg01003813     chrX   2715058      *</span></span>
<span><span class="co">#&gt;   cg03723195     chrX   2847354      *</span></span>
<span><span class="co">#&gt;   cg00074638     chrX   2883307      *</span></span>
<span><span class="co">#&gt;   cg01728536     chrX   7070451      *</span></span>
<span><span class="co">#&gt;   cg01864404     chrX   8434367      *</span></span>
<span><span class="co">#&gt;          ...      ...       ...    ...</span></span>
<span><span class="co">#&gt;   cg26983430     chrY  24549675      *</span></span>
<span><span class="co">#&gt;   cg01757887     chrY  25114810      *</span></span>
<span><span class="co">#&gt;   cg00061679     chrY  25314171      *</span></span>
<span><span class="co">#&gt;   cg03930849     chrY  26716289      *</span></span>
<span><span class="co">#&gt;   cg08265308     chrY  28555550      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 2 sequences from hg19 genome; no seqlengths</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="summarizing-gene-region-types">Summarizing Gene Region Types<a class="anchor" aria-label="anchor" href="#summarizing-gene-region-types"></a>
</h3>
<div class="section level4">
<h4 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h4>
<p>Gene regions, which include functional segments such as promoters,
gene bodies, and intergenic regions, play pivotal roles in gene
expression and regulation. Summarizing methylation patterns across these
regions can provide insights into potential gene regulatory mechanisms
and associations with phenotypes or disease states. Herein, we will
delve into how to succinctly summarize methylation data at these crucial
genomic segments using the <code>regionalpcs</code> package.</p>
</div>
<div class="section level4">
<h4 id="load-gene-region-annotations">Load Gene Region Annotations<a class="anchor" aria-label="anchor" href="#load-gene-region-annotations"></a>
</h4>
<p>First, letâ€™s load the gene region annotations. Make sure to align the
genomic builds of your annotations and methylation data.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Obtain promoter regions</span></span>
<span><span class="co"># The TxDb object 'txdb' facilitates the retrieval of transcript-based </span></span>
<span><span class="co"># genomic annotations.</span></span>
<span><span class="va">txdb</span> <span class="op">&lt;-</span> <span class="va">TxDb.Hsapiens.UCSC.hg19.knownGene</span></span>
<span></span>
<span><span class="co"># Extracting promoter regions with a defined upstream and downstream window. </span></span>
<span><span class="co"># This GRanges object 'promoters_gr' will be utilized to map and summarize </span></span>
<span><span class="co"># methylation data in promoter regions.</span></span>
<span><span class="va">promoters_gr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/intra-range-methods.html" class="external-link">promoters</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span>, </span>
<span>                                    upstream<span class="op">=</span><span class="fl">1000</span>, </span>
<span>                                    downstream<span class="op">=</span><span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Display the GRanges object containing the genomic coordinates of promoter </span></span>
<span><span class="co"># regions.</span></span>
<span><span class="va">promoters_gr</span></span>
<span><span class="co">#&gt; GRanges object with 23056 ranges and 1 metadata column:</span></span>
<span><span class="co">#&gt;         seqnames              ranges strand |     gene_id</span></span>
<span><span class="co">#&gt;            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">#&gt;       1    chr19   58874215-58875214      - |           1</span></span>
<span><span class="co">#&gt;      10     chr8   18247755-18248754      + |          10</span></span>
<span><span class="co">#&gt;     100    chr20   43280377-43281376      - |         100</span></span>
<span><span class="co">#&gt;    1000    chr18   25757446-25758445      - |        1000</span></span>
<span><span class="co">#&gt;   10000     chr1 244006887-244007886      - |       10000</span></span>
<span><span class="co">#&gt;     ...      ...                 ...    ... .         ...</span></span>
<span><span class="co">#&gt;    9991     chr9 115095945-115096944      - |        9991</span></span>
<span><span class="co">#&gt;    9992    chr21   35735323-35736322      + |        9992</span></span>
<span><span class="co">#&gt;    9993    chr22   19109968-19110967      - |        9993</span></span>
<span><span class="co">#&gt;    9994     chr6   90538619-90539618      + |        9994</span></span>
<span><span class="co">#&gt;    9997    chr22   50964906-50965905      - |        9997</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 93 sequences (1 circular) from hg19 genome</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="create-a-region-map">Create a Region Map<a class="anchor" aria-label="anchor" href="#create-a-region-map"></a>
</h4>
<p>Creating a region map, which systematically assigns CpGs to specific
gene regions, stands as a crucial precursor to gene-region summarization
using the <code>regionalpcs</code> package. This mapping elucidates the
physical positioning of CpGs within particular gene regions,
facilitating our upcoming endeavors to comprehend how methylation varies
across distinct genomic segments. Weâ€™ll use the
<code>create_region_map</code> function from the
<code>regionalpcs</code> package. This function takes two genomic ranges
objects, <code>cpg_gr</code> contains CpG positions and
<code>genes_gr</code> contains gene region positions. Make sure both
positions are aligned to the same genome build (e.g.Â GrCH37,
CrCH38).</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># get the region map using the regionalpcs function</span></span>
<span><span class="va">region_map</span> <span class="op">&lt;-</span> <span class="fu">regionalpcs</span><span class="fu">::</span><span class="fu"><a href="../reference/create_region_map.html">create_region_map</a></span><span class="op">(</span>cpg_gr<span class="op">=</span><span class="va">cpg_gr</span>, </span>
<span>                                                genes_gr<span class="op">=</span><span class="va">promoters_gr</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Display the initial few rows of the region map.</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">region_map</span><span class="op">)</span></span>
<span><span class="co">#&gt;     gene_id     cpg_id</span></span>
<span><span class="co">#&gt; 1 100131434 cg00466309</span></span>
<span><span class="co">#&gt; 2 100133941 cg05230942</span></span>
<span><span class="co">#&gt; 3 100133957 cg00636562</span></span>
<span><span class="co">#&gt; 4     10084 cg01303569</span></span>
<span><span class="co">#&gt; 5      1069 cg01056373</span></span>
<span><span class="co">#&gt; 6     10857 cg01333849</span></span></code></pre></div>
<p><strong>Note:</strong> The second column of <code>region_map</code>
must contain values matching the rownames of your methylation
dataframe.</p>
</div>
<div class="section level4">
<h4 id="summarizing-gene-regions-with-regional-principal-components">Summarizing Gene Regions with Regional Principal Components<a class="anchor" aria-label="anchor" href="#summarizing-gene-regions-with-regional-principal-components"></a>
</h4>
<p>In this final section, weâ€™ll summarize gene regions using Principal
Components (PCs) to capture the maximum variation. Weâ€™ll utilize the
<code>compute_regional_pcs</code> function from the
<code>regionalpcs</code> package for this.</p>
<div class="section level5">
<h5 id="compute-regional-pcs">Compute Regional PCs<a class="anchor" aria-label="anchor" href="#compute-regional-pcs"></a>
</h5>
<p>Letâ€™s calculate the regional PCs using our gene regions for
demonstration purposes.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Display head of region map</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">region_map</span><span class="op">)</span></span>
<span><span class="co">#&gt;     gene_id     cpg_id</span></span>
<span><span class="co">#&gt; 1 100131434 cg00466309</span></span>
<span><span class="co">#&gt; 2 100133941 cg05230942</span></span>
<span><span class="co">#&gt; 3 100133957 cg00636562</span></span>
<span><span class="co">#&gt; 4     10084 cg01303569</span></span>
<span><span class="co">#&gt; 5      1069 cg01056373</span></span>
<span><span class="co">#&gt; 6     10857 cg01333849</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">region_map</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 211   2</span></span>
<span></span>
<span><span class="co"># Compute regional PCs</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_regional_pcs.html">compute_regional_pcs</a></span><span class="op">(</span>meth<span class="op">=</span><span class="va">mvals</span>, region_map<span class="op">=</span><span class="va">region_map</span>, pc_method<span class="op">=</span><span class="st">"gd"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using Gavish-Donoho method</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="inspecting-the-output">Inspecting the Output<a class="anchor" aria-label="anchor" href="#inspecting-the-output"></a>
</h5>
<p>The function returns a list containing multiple elements. Letâ€™s first
look at what these elements are.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Inspect the output list elements</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "regional_pcs"     "percent_variance" "loadings"         "info"</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="extracting-and-viewing-regional-pcs">Extracting and Viewing Regional PCs<a class="anchor" aria-label="anchor" href="#extracting-and-viewing-regional-pcs"></a>
</h5>
<p>The first element (<code>res$regional_pcs</code>) is a data frame
containing the calculated regional PCs.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Extract regional PCs</span></span>
<span><span class="va">regional_pcs</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="va">regional_pcs</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">regional_pcs</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">4</span><span class="op">]</span></span>
<span><span class="co">#&gt;               5723646052_R02C02 5723646052_R04C01 5723646052_R05C02</span></span>
<span><span class="co">#&gt; 100131434-PC1         -3.887949        1.82168108        -4.2354365</span></span>
<span><span class="co">#&gt; 100133941-PC1          1.100370        0.06719947         0.1932361</span></span>
<span><span class="co">#&gt; 100133957-PC1         -2.032026        0.19154436        -0.9193417</span></span>
<span><span class="co">#&gt; 10084-PC1             -5.011579        2.34574585        -4.2050727</span></span>
<span><span class="co">#&gt; 1069-PC1              -2.004785        1.20363622        -2.5436686</span></span>
<span><span class="co">#&gt; 10857-PC1             -1.986294        1.31482789        -2.0044911</span></span>
<span><span class="co">#&gt;               5723646053_R04C02</span></span>
<span><span class="co">#&gt; 100131434-PC1         2.1707679</span></span>
<span><span class="co">#&gt; 100133941-PC1        -0.7438658</span></span>
<span><span class="co">#&gt; 100133957-PC1         1.1903709</span></span>
<span><span class="co">#&gt; 10084-PC1             2.2586300</span></span>
<span><span class="co">#&gt; 1069-PC1              1.5851359</span></span>
<span><span class="co">#&gt; 10857-PC1             2.1369556</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="understanding-the-results">Understanding the Results<a class="anchor" aria-label="anchor" href="#understanding-the-results"></a>
</h5>
<p>The output is a data frame with regional PCs for each region as rows
and our samples as columns. This is our new representation of
methylation values, now on a gene regional PC scale. We can feed these
into downstream analyses as is.</p>
<p>The number of regional PCs representing each gene region was
determined by the Gavish-Donoho method. This method allows us to
identify PCs that capture actual signal in our data and not the noise
that is inherent in any dataset. To explore alternative methods, we can
change the <code>pc_method</code> parameter.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Count the number of unique gene regions and PCs</span></span>
<span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>gene_pc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">regional_pcs</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span><span class="va">gene_pc</span>, into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"pc"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"-"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">regions</span><span class="op">)</span></span>
<span><span class="co">#&gt;        gene  pc</span></span>
<span><span class="co">#&gt; 1 100131434 PC1</span></span>
<span><span class="co">#&gt; 2 100133941 PC1</span></span>
<span><span class="co">#&gt; 3 100133957 PC1</span></span>
<span><span class="co">#&gt; 4     10084 PC1</span></span>
<span><span class="co">#&gt; 5      1069 PC1</span></span>
<span><span class="co">#&gt; 6     10857 PC1</span></span>
<span></span>
<span><span class="co"># number of genes that have been summarized</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">regions</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 142</span></span>
<span></span>
<span><span class="co"># how many of each PC did we get</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">regions</span><span class="op">$</span><span class="va">pc</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; PC1 </span></span>
<span><span class="co">#&gt; 142</span></span></code></pre></div>
<p>We have summarized each of our genes using just one PC. The number of
PCs depends on three main factors: the number of samples, the number of
CpGs in the gene region, and the noise in the methylation data.</p>
<p>By default, the <code>compute_regional_pcs</code> function uses the
Gavish-Donoho method. However, we can also use the Marcenko-Pasteur
method by setting the <code>pc_method</code> parameter:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Using Marcenko-Pasteur method</span></span>
<span><span class="va">mp_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/compute_regional_pcs.html">compute_regional_pcs</a></span><span class="op">(</span><span class="va">mvals</span>, <span class="va">region_map</span>, pc_method <span class="op">=</span> <span class="st">"mp"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Using Marchenko-Pastur method</span></span>
<span></span>
<span><span class="co"># select the regional pcs</span></span>
<span><span class="va">mp_regional_pcs</span> <span class="op">&lt;-</span> <span class="va">mp_res</span><span class="op">$</span><span class="va">regional_pcs</span></span>
<span></span>
<span><span class="co"># separate the genes from the pc numbers</span></span>
<span><span class="va">mp_regions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>gene_pc <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">mp_regional_pcs</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span><span class="va">gene_pc</span>, into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gene"</span>, <span class="st">"pc"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"-"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mp_regions</span><span class="op">)</span></span>
<span><span class="co">#&gt;        gene  pc</span></span>
<span><span class="co">#&gt; 1 100131434 PC1</span></span>
<span><span class="co">#&gt; 2 100133941 PC1</span></span>
<span><span class="co">#&gt; 3 100133957 PC1</span></span>
<span><span class="co">#&gt; 4     10084 PC1</span></span>
<span><span class="co">#&gt; 5      1069 PC1</span></span>
<span><span class="co">#&gt; 6     10857 PC1</span></span>
<span></span>
<span><span class="co"># number of genes that have been summarized</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">mp_regions</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 142</span></span>
<span></span>
<span><span class="co"># how many of each PC did we get</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">mp_regions</span><span class="op">$</span><span class="va">pc</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; PC1 </span></span>
<span><span class="co">#&gt; 142</span></span></code></pre></div>
<p>The Marcenko-Pasteur and the Gavish-Donoho methods are both based on
Random Matrix Theory, and they aim to identify the number of significant
PCs that capture the true signal in the data and not just the noise.
However, these methods differ in how they select the number of
significant PCs. The Marcenko-Pasteur method typically selects more PCs
to represent a gene region compared to the Gavish-Donoho method. This
may be due to the different ways in which the two methods estimate the
noise level in the data.</p>
<p>Ultimately, the choice between the two methods depends on the
specific needs and goals of the analysis. The Gavish-Donoho method tends
to provide more conservative results, while the Marcenko-Pasteur method
may capture more of the underlying signal in the data. Researchers
should carefully consider their objectives and the characteristics of
their data when selecting a method.</p>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session Information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.3.1 (2023-06-16 ucrt)</span></span>
<span><span class="co">#&gt; Platform: x86_64-w64-mingw32/x64 (64-bit)</span></span>
<span><span class="co">#&gt; Running under: Windows 10 x64 (build 19045)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] LC_COLLATE=English_United States.utf8 </span></span>
<span><span class="co">#&gt; [2] LC_CTYPE=English_United States.utf8   </span></span>
<span><span class="co">#&gt; [3] LC_MONETARY=English_United States.utf8</span></span>
<span><span class="co">#&gt; [4] LC_NUMERIC=C                          </span></span>
<span><span class="co">#&gt; [5] LC_TIME=English_United States.utf8    </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/Los_Angeles</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] parallel  stats4    stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">#&gt; [8] methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] TxDb.Hsapiens.UCSC.hg19.knownGene_3.2.2           </span></span>
<span><span class="co">#&gt;  [2] GenomicFeatures_1.53.1                            </span></span>
<span><span class="co">#&gt;  [3] AnnotationDbi_1.63.2                              </span></span>
<span><span class="co">#&gt;  [4] minfiData_0.47.0                                  </span></span>
<span><span class="co">#&gt;  [5] IlluminaHumanMethylation450kanno.ilmn12.hg19_0.6.1</span></span>
<span><span class="co">#&gt;  [6] IlluminaHumanMethylation450kmanifest_0.4.0        </span></span>
<span><span class="co">#&gt;  [7] minfi_1.47.0                                      </span></span>
<span><span class="co">#&gt;  [8] bumphunter_1.43.0                                 </span></span>
<span><span class="co">#&gt;  [9] locfit_1.5-9.8                                    </span></span>
<span><span class="co">#&gt; [10] iterators_1.0.14                                  </span></span>
<span><span class="co">#&gt; [11] foreach_1.5.2                                     </span></span>
<span><span class="co">#&gt; [12] Biostrings_2.69.2                                 </span></span>
<span><span class="co">#&gt; [13] XVector_0.41.1                                    </span></span>
<span><span class="co">#&gt; [14] SummarizedExperiment_1.31.1                       </span></span>
<span><span class="co">#&gt; [15] Biobase_2.61.0                                    </span></span>
<span><span class="co">#&gt; [16] MatrixGenerics_1.13.1                             </span></span>
<span><span class="co">#&gt; [17] matrixStats_1.0.0                                 </span></span>
<span><span class="co">#&gt; [18] dplyr_1.1.2                                       </span></span>
<span><span class="co">#&gt; [19] tibble_3.2.1                                      </span></span>
<span><span class="co">#&gt; [20] tidyr_1.3.0                                       </span></span>
<span><span class="co">#&gt; [21] GenomicRanges_1.53.1                              </span></span>
<span><span class="co">#&gt; [22] GenomeInfoDb_1.37.2                               </span></span>
<span><span class="co">#&gt; [23] IRanges_2.35.2                                    </span></span>
<span><span class="co">#&gt; [24] S4Vectors_0.39.1                                  </span></span>
<span><span class="co">#&gt; [25] BiocGenerics_0.47.0                               </span></span>
<span><span class="co">#&gt; [26] regionalpcs_0.99.4                                </span></span>
<span><span class="co">#&gt; [27] BiocStyle_2.29.1                                  </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] splines_4.3.1             BiocIO_1.11.0            </span></span>
<span><span class="co">#&gt;   [3] bitops_1.0-7              filelock_1.0.2           </span></span>
<span><span class="co">#&gt;   [5] preprocessCore_1.63.1     XML_3.99-0.14            </span></span>
<span><span class="co">#&gt;   [7] lifecycle_1.0.3           rprojroot_2.0.3          </span></span>
<span><span class="co">#&gt;   [9] lattice_0.21-8            MASS_7.3-60              </span></span>
<span><span class="co">#&gt;  [11] base64_2.0.1              scrime_1.3.5             </span></span>
<span><span class="co">#&gt;  [13] magrittr_2.0.3            limma_3.57.7             </span></span>
<span><span class="co">#&gt;  [15] sass_0.4.7                rmarkdown_2.24           </span></span>
<span><span class="co">#&gt;  [17] jquerylib_0.1.4           yaml_2.3.7               </span></span>
<span><span class="co">#&gt;  [19] doRNG_1.8.6               askpass_1.1              </span></span>
<span><span class="co">#&gt;  [21] cowplot_1.1.1             DBI_1.1.3                </span></span>
<span><span class="co">#&gt;  [23] RColorBrewer_1.1-3        abind_1.4-5              </span></span>
<span><span class="co">#&gt;  [25] zlibbioc_1.47.0           quadprog_1.5-8           </span></span>
<span><span class="co">#&gt;  [27] purrr_1.0.2               RCurl_1.98-1.12          </span></span>
<span><span class="co">#&gt;  [29] rappdirs_0.3.3            GenomeInfoDbData_1.2.10  </span></span>
<span><span class="co">#&gt;  [31] RMTstat_0.3.1             ggrepel_0.9.3            </span></span>
<span><span class="co">#&gt;  [33] irlba_2.3.5.1             genefilter_1.83.1        </span></span>
<span><span class="co">#&gt;  [35] dqrng_0.3.0               annotate_1.79.0          </span></span>
<span><span class="co">#&gt;  [37] pkgdown_2.0.7             DelayedMatrixStats_1.23.4</span></span>
<span><span class="co">#&gt;  [39] codetools_0.2-19          DelayedArray_0.27.10     </span></span>
<span><span class="co">#&gt;  [41] xml2_1.3.5                tidyselect_1.2.0         </span></span>
<span><span class="co">#&gt;  [43] ScaledMatrix_1.9.1        beanplot_1.3.1           </span></span>
<span><span class="co">#&gt;  [45] BiocFileCache_2.9.1       illuminaio_0.43.0        </span></span>
<span><span class="co">#&gt;  [47] GenomicAlignments_1.37.0  jsonlite_1.8.7           </span></span>
<span><span class="co">#&gt;  [49] multtest_2.57.0           survival_3.5-7           </span></span>
<span><span class="co">#&gt;  [51] systemfonts_1.0.4         tools_4.3.1              </span></span>
<span><span class="co">#&gt;  [53] progress_1.2.2            ragg_1.2.5               </span></span>
<span><span class="co">#&gt;  [55] Rcpp_1.0.11               glue_1.6.2               </span></span>
<span><span class="co">#&gt;  [57] SparseArray_1.1.11        xfun_0.40                </span></span>
<span><span class="co">#&gt;  [59] HDF5Array_1.29.3          withr_2.5.0              </span></span>
<span><span class="co">#&gt;  [61] BiocManager_1.30.22       fastmap_1.1.1            </span></span>
<span><span class="co">#&gt;  [63] rhdf5filters_1.13.5       fansi_1.0.4              </span></span>
<span><span class="co">#&gt;  [65] openssl_2.1.0             rsvd_1.0.5               </span></span>
<span><span class="co">#&gt;  [67] digest_0.6.33             R6_2.5.1                 </span></span>
<span><span class="co">#&gt;  [69] textshaping_0.3.6         colorspace_2.1-0         </span></span>
<span><span class="co">#&gt;  [71] biomaRt_2.57.1            RSQLite_2.3.1            </span></span>
<span><span class="co">#&gt;  [73] utf8_1.2.3                generics_0.1.3           </span></span>
<span><span class="co">#&gt;  [75] data.table_1.14.8         rtracklayer_1.61.1       </span></span>
<span><span class="co">#&gt;  [77] prettyunits_1.1.1         httr_1.4.7               </span></span>
<span><span class="co">#&gt;  [79] S4Arrays_1.1.5            pkgconfig_2.0.3          </span></span>
<span><span class="co">#&gt;  [81] gtable_0.3.3              blob_1.2.4               </span></span>
<span><span class="co">#&gt;  [83] siggenes_1.75.0           htmltools_0.5.6          </span></span>
<span><span class="co">#&gt;  [85] bookdown_0.35             scales_1.2.1             </span></span>
<span><span class="co">#&gt;  [87] png_0.1-8                 knitr_1.43               </span></span>
<span><span class="co">#&gt;  [89] rstudioapi_0.15.0         reshape2_1.4.4           </span></span>
<span><span class="co">#&gt;  [91] tzdb_0.4.0                rjson_0.2.21             </span></span>
<span><span class="co">#&gt;  [93] nlme_3.1-163              curl_5.0.2               </span></span>
<span><span class="co">#&gt;  [95] cachem_1.0.8              rhdf5_2.45.1             </span></span>
<span><span class="co">#&gt;  [97] stringr_1.5.0             restfulr_0.0.15          </span></span>
<span><span class="co">#&gt;  [99] desc_1.4.2                GEOquery_2.69.0          </span></span>
<span><span class="co">#&gt; [101] pillar_1.9.0              grid_4.3.1               </span></span>
<span><span class="co">#&gt; [103] reshape_0.8.9             vctrs_0.6.3              </span></span>
<span><span class="co">#&gt; [105] BiocSingular_1.17.1       beachmat_2.17.15         </span></span>
<span><span class="co">#&gt; [107] dbplyr_2.3.3              xtable_1.8-4             </span></span>
<span><span class="co">#&gt; [109] evaluate_0.21             readr_2.1.4              </span></span>
<span><span class="co">#&gt; [111] cli_3.6.1                 compiler_4.3.1           </span></span>
<span><span class="co">#&gt; [113] Rsamtools_2.17.0          rlang_1.1.1              </span></span>
<span><span class="co">#&gt; [115] crayon_1.5.2              rngtools_1.5.2           </span></span>
<span><span class="co">#&gt; [117] nor1mix_1.3-0             mclust_6.0.0             </span></span>
<span><span class="co">#&gt; [119] plyr_1.8.8                fs_1.6.3                 </span></span>
<span><span class="co">#&gt; [121] stringi_1.7.12            BiocParallel_1.35.3      </span></span>
<span><span class="co">#&gt; [123] munsell_0.5.0             PCAtools_2.13.0          </span></span>
<span><span class="co">#&gt; [125] Matrix_1.6-1              hms_1.1.3                </span></span>
<span><span class="co">#&gt; [127] sparseMatrixStats_1.13.4  bit64_4.0.5              </span></span>
<span><span class="co">#&gt; [129] ggplot2_3.4.3             Rhdf5lib_1.23.0          </span></span>
<span><span class="co">#&gt; [131] KEGGREST_1.41.0           statmod_1.5.0            </span></span>
<span><span class="co">#&gt; [133] memoise_2.0.1             bslib_0.5.1              </span></span>
<span><span class="co">#&gt; [135] bit_4.0.5</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Tiffany Eulalio.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
